packages/vuetify/src/components/Vuetify/mixins/breakpoint.ts

```diff
@@ -1,121 +1,137 @@
 import Vue from 'vue'
-import { VuetifyBreakpoint } from 'vuetify/types'
+import { VuetifyUseOptions, VuetifyBreakpoint } from 'vuetify/types'
 
 /**
- * A modified version of https://gist.github.com/cb109/b074a65f7595cffc21cea59ce8d15f9b
+ * Factory function for the breakpoint mixin.
  */
+export default function breakpoint (opts: VuetifyUseOptions['breakpoint'] = {}) {
+  if (!opts) {
+    opts = {}
+  }
 
-/**
- * A Vue mixin to get the current width/height and the associated breakpoint.
- *
- *   <div v-if="$breakpoint.smAndDown">...</div>
- *
- */
-export default Vue.extend({
-  data: () => ({
-    clientHeight: getClientHeight(),
-    clientWidth: getClientWidth(),
-    resizeTimeout: undefined as number | undefined
-  }),
-
-  computed: {
-    breakpoint (): VuetifyBreakpoint {
-      const xs = this.clientWidth < 600
-      const sm = this.clientWidth < 960 && !xs
-      const md = this.clientWidth < (1280 - 16) && !(sm || xs)
-      const lg = this.clientWidth < (1920 - 16) && !(md || sm || xs)
-      const xl = this.clientWidth >= (1920 - 16)
-
-      const xsOnly = xs
-      const smOnly = sm
-      const smAndDown = (xs || sm) && !(md || lg || xl)
-      const smAndUp = !xs && (sm || md || lg || xl)
-      const mdOnly = md
-      const mdAndDown = (xs || sm || md) && !(lg || xl)
-      const mdAndUp = !(xs || sm) && (md || lg || xl)
-      const lgOnly = lg
-      const lgAndDown = (xs || sm || md || lg) && !xl
-      const lgAndUp = !(xs || sm || md) && (lg || xl)
-      const xlOnly = xl
-
-      let name
-      switch (true) {
-        case (xs):
-          name = 'xs'
-          break
-        case (sm):
-          name = 'sm'
-          break
-        case (md):
-          name = 'md'
-          break
-        case (lg):
-          name = 'lg'
-          break
-        default:
-          name = 'xl'
-          break
-      }
+  const DEFAULTS_FOR_USER_CONFIGURABLE_OPTIONS = {
```

Comment:
Rename to `BREAKPOINTS_DEFAULTS` and move outside the function body.

Reply:
Done